packages/react/src/components/AccountSettings/DeleteUser/DeleteUser.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+import { Logger } from 'aws-amplify';
+import { deleteUser, translate } from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { Flex } from '../../../primitives';
+import {
+  DefaultWarning,
+  DefaultError,
+  DefaultSubmitButton,
+} from './defaultComponents';
+import { DeleteUserProps, DeleteUserState } from './types';
+
+const logger = new Logger('DeleteUser');
+
+function DeleteUser({
+  onSuccess,
+  onError,
+  handleDelete,
+}: DeleteUserProps): JSX.Element | null {
+  const [state, setState] = React.useState<DeleteUserState>('IDLE');
+  const [errorMessage, setErrorMessage] = React.useState<string>(null);
+
+  // translations
+  const deleteAccountText = translate('Delete Account');
+
+  const { user, isLoading } = useAuth();
+
+  const startConfirmation = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setState('IS_CONFIRMING');
+  };
+
+  const runDeleteUser = React.useCallback(async () => {
+    setState('IN_PROGRESS');
+    try {
+      if (handleDelete) {
+        await handleDelete(user);
+      } else {
+        await deleteUser();
+      }
+      setState('DONE');
+      onSuccess?.();
+    } catch (e) {
+      const error = e as Error;
```

Comment:
Is this cast necessary to access `error.message`?

Reply:
Right, and also because `onError(error)` expects `error` to be of `Error` type.

Having to type `error` from catch clause is a limitation with TS: https://github.com/microsoft/TypeScript/issues/20024